internal/backend/projects/lint.go

```diff
@@ -0,0 +1,108 @@
+package projects
+
+import (
+	"fmt"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/projectsgrpcsvc"
+	"go.autokitteh.dev/autokitteh/internal/manifest"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+type Checker func(projectID sdktypes.ProjectID, manifest *manifest.Manifest, resources map[string][]byte) []*sdktypes.CheckViolation
+
+var (
+	lintCheckers []Checker
+)
+
+const manifestFile = "autokitteh.yaml"
+
+func init() {
+	lintCheckers = []Checker{
+		checkNoTriggers,
+		checkEmptyVars,
+	}
+}
+
+func Validate(projectID sdktypes.ProjectID, manifest *manifest.Manifest, resources map[string][]byte) []*sdktypes.CheckViolation {
+	var vs []*sdktypes.CheckViolation
+	for _, checker := range lintCheckers {
+		vs = append(vs, checker(projectID, manifest, resources)...)
+	}
+
+	return vs
+}
+
+func checkNoTriggers(_ sdktypes.ProjectID, m *manifest.Manifest, _ map[string][]byte) []*sdktypes.CheckViolation {
+	if len(m.Project.Triggers) == 0 {
```

Comment:
if manifest is nil here it's gonna panic
better check that it is not nil or add a checkManifestProvidedChecker before

Reply:
If manifest is `nil` we should panic

Reply:
this is backend code
you can't panic

Reply:
I can panic in a handler, gRPC recovers.